src/datasets/dataset_dict.py

```diff
@@ -795,7 +797,7 @@ def map(
         writer_batch_size: Optional[int] = 1000,
         features: Optional[Features] = None,
         disable_nullable: bool = False,
-        fn_kwargs: Optional[dict] = None,
+        fn_kwargs: dict = {},
```

Comment:
I think you can revert this change and keep it as None by default :)

It's also good practice to avoid having lists or dicts as default parameters, since they can be modified inadvertently